docker/package/wizard_structure.py

```diff
@@ -18,6 +18,7 @@
 protocol_hash_regex = (
     b"P[123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz]{50}"
 )
+signer_uri_regex = b"((?:tcp|unix|https|http):\/\/.+)\/keys\/(tz[123]\w{33})\/?"
```

Comment:
Seems like this regex is too strict:
```
Provide your remote key with the address of the signer.
> tcp://localhost:7732/tz1QMRfsLZUyyHVthxPwoKgCyubSG1RWoLnV
Validation error: The input doesn't match the format for a remote signer URI: (tcp|unix|https|http)://<host address>/<public key address>
Please check the input and try again.
```

Reply:
Looks like `keys/` part is actually redundant for key importing:
```
sudo -u tezos tezos-client import secret key remote tcp://localhost:7732/keys/tz1atFdTEzZPDvq4XRpghYQ9g3SrXUHvduSC --force
Error:
  Cannot parse the key uri: tcp://localhost:7732/keys/tz1atFdTEzZPDvq4XRpghYQ9g3SrXUHvduSC
  Failed to read a b58check_encoding data (Signature.Public_key_hash): "keys/tz1atFdTEzZPDvq4XRpghYQ9g3SrXUHvduSC"
```

Reply:
Oh, I think I understand now. This regex doesn't follow the remote URL format specified by `tezos-client` and is used to determine the key and the host separately. @ixahedron, do I understand it correctly?

Reply:
🤔 I only tested it with `http`, but when running e.g. `tezos-client -R http://127.0.0.1:6732 import secret key baker remote:tz1atFdTEzZPDvq4XRpghYQ9g3SrXUHvduSC`, it actually tries to connect to `http://127.0.0.1:6732/keys/tz1atFdTEzZPDvq4XRpghYQ9g3SrXUHvduSC`, with the `keys/` part. It's also the format you used as an example in the original issue. I assumed this is the format one will use when sharing key addresses, too. So, no, my intention wasn't to invent a new format 🙈

Reply:
hmm, seems like different signing schemes have different key URLs then :confused:

Reply:
Surprisingly `tcp://localhost:7732/keys/tz1atFdTEzZPDvq4XRpghYQ9g3SrXUHvduSC` works with the wizard, but doesn't work with `tezos-client`

Reply:
> tcp://localhost:7732/keys/tz1atFdTEzZPDvq4XRpghYQ9g3SrXUHvduSC works with the wizard
Yeah, 'cause it indeed separates the URI into the host and the key to use with the supposedly more flexible `-R` flag, throwing out the `keys/` part.

Reply:
I couldn't find documentation on the signer URI format btw 😕

Reply:
Neither do I

Reply:
```
$ sudo -u tezos tezos-client import secret key remote http://localhost:6732/keys/tz1atFdTEzZPDvq4XRpghYQ9g3SrXUHvduSC --force
Error:
  Rpc request failed:
     - meth: GET
     - uri: http://localhost:6732/keys/keys/tz1atFdTEzZPDvq4XRpghYQ9g3SrXUHvduSC
     - error: Oops! It looks like we forged an invalid HTTP request.
                Failed to parsed an argument in path. After "keys/keys", the value "failed to decode Base58Check-encoded data (Signature.Public_key_hash): "keys"" is not acceptable for type "pkh"
```

Hmm, apparently, I provided the wrong URI example initially :see_no_evil:

Reply:
And `keys/` part is actually added by the `tezos-client` implicitly

Reply:
Let's stick with the format used by the `tezos-client import secret key` command to be more convenient for `tezos-signer` users